src/Symfony/Component/Console/Command/LockableTrait.php

```diff
@@ -26,6 +26,17 @@ trait LockableTrait
 {
     private ?LockInterface $lock = null;
 
+    private ?LockFactory $lockFactory = null;
+
+    public function setLockFactory(?LockFactory $lockFactory): void
+    {
+        if (null !== $this->lock) {
+            throw new LogicException('You cannot set the lockFactory if a lock is already in place.');
+        }
+
+        $this->lockFactory = $lockFactory;
+    }
```

Comment:
Why do we need a public setter? Wouldn't I want to inject the factory via the command's constructor instead?

Reply:
> Why do we need a public setter? Wouldn't I want to inject the factory via the command's constructor instead?

I don't know, I wasn't sure about the best way to "configure" the lockFactory in the trait.

My first idea was to add a setter `setLockFactory` with a `@required` annotation in other to avoid the need to inject the factory with the command constructor. But I think this would be a BC break since it will change the behavior of the trait ; so I removed the `@required` annotation.

I don't use trait often and didn't thought that in
```
use LockableTrait;

public function __construct(
    private readonly LockFactory $lockFactory
) {
    $this->setLockFactory($lockFactory);
}
```
The setter call is useless because the trait share his properties with the class.

I'll remove it.
My dream solution would be to avoid the need of injecting the factory with the command constructor... (just injecting the trait would be enough) but I'm not sure it's possible.